### PR TITLE
Fixes #8073: Support for more Debian / Ubuntu repository components

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8073

### CODE REVIEW

File: techniques/applications/aptPackageManagerSettings/3.0/metadata.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@ appreciated by administrators on the APT package
 manager, and eventually enforce them.
 -->
 <TECHNIQUE name="APT package manager configuration">
+  <DEPRECATED>This technique version has been superseded by a new version. It will no longer be available in the next stable version of Rudder. Please upgrade to the latest version.</DEPRECATED>
```
Comment (ncharles) [inline techniques/applications/aptPackageManagerSettings/3.0/metadata.xml]: are you sure you already want to deprecate it ?
shouldn't it be deprecated in 3.3 ?

File: techniques/applications/aptPackageManagerSettings/4.0/aptPackageManagerSettings.st [hunk 2]
```diff
@@ -122,7 +122,7 @@ bundle agent check_apt_settings
     &APT_URL:{apt_url |"apt_url[&i&]" string => "&apt_url&";
 }&
 
-    &APT_COMPONENT_MAIN, APT_COMPONENT_CONTRIB, APT_COMPONENT_NON_FREE, APT_COMPONENT_RESTRICTED, APT_COMPONENT_UNIVERSE, APT_COMPONENT_MULTIVERSE:{main, contrib, nonfree, restricted, universe, multiverse |"apt_components[&i&]" string => "&main& &contrib& &nonfree& &restricted& &universe& &multiverse&";
+    &APT_COMPONENT_MAIN, APT_COMPONENT_CONTRIB, APT_COMPONENT_NON_FREE, APT_COMPONENT_RESTRICTED, APT_COMPONENT_UNIVERSE, APT_COMPONENT_MULTIVERSE:{main, contrib, nonfree, restricted, universe, multiverse |"apt_components[&i&]" string => "&main& &contrib& &nonfree& &restricted& &universe& &multiverse& &APT_COMPONENT_OTHER&";
```
Comment (ncharles) [inline techniques/applications/aptPackageManagerSettings/4.0/aptPackageManagerSettings.st]: i'm pretty sure this syntax is invalid
it should be

&APT_COMPONENT_MAIN, APT_COMPONENT_CONTRIB, APT_COMPONENT_NON_FREE, APT_COMPONENT_RESTRICTED, APT_COMPONENT_UNIVERSE, APT_COMPONENT_MULTIVERSE, APT_COMPONENT_OTHER:{main, contrib, nonfree, restricted, universe, multiverse, other |"apt_components[&i&]" string => "&main& &contrib& &nonfree& &restricted& &universe& &multiverse& &other&";

**Comment by amousset** [comment]
PR updated

**Comment by amousset** [comment]
PR updated.

**Comment by amousset** [comment]
PR updated

**Comment by ncharles** [comment]
thank you !

**Pull request merged**